src/mmw/apps/modeling/calcs.py

```diff
@@ -92,3 +92,42 @@ def point_source_pollution(geojson):
         'name': 'pointsource',
         'categories': point_source_results
     }
+
+
+def catchment_water_quality(geojson):
+    """
+    Given a GeoJSON shape, retrieve Catchment Water Quality data
+    from the `drb_catchment_water_quality` table to display
+    in the Analyze tab.
+
+    Returns a dictionary to append to the outgoing JSON for analysis
+    result
+    """
+    geom = GEOSGeometry(geojson, srid=4326)
+    table_name = 'drb_catchment_water_quality'
+    sql = '''
+    SELECT nord, areaha, tn_tot_kgy, tp_tot_kgy, tss_tot_kg,
+        tn_urban_k, tn_riparia, tn_ag_kgyr, tn_natural, tn_pt_kgyr,
+        tp_urban_k, tp_riparia, tp_ag_kgyr, tp_natural, tp_pt_kgyr,
+        tss_urban_, tss_rip_kg, tss_ag_kgy, tss_natura,
+        ST_AsGeoJSON(geom) as geom
+    FROM {table_name}
+    WHERE ST_Intersects(geom, ST_SetSRID(ST_GeomFromText(%s), 4326))
+    '''.format(table_name=table_name)
+
+    with connection.cursor() as cursor:
+        cursor.execute(sql, [geom.wkt])
+
+        if cursor.rowcount != 0:
+            columns = [col[0] for col in cursor.description]
+            catchment_water_quality_results = [
+                dict(zip(columns, row))
```

Comment:
It looks like the point source function does some coalescing of the return values to replace null values with `None`: https://github.com/WikiWatershed/model-my-watershed/blob/d927dbb712265254982c87b33d8308e9d0eac5e7/src/mmw/apps/modeling/calcs.py#L82

I think you've got some checks for that on the frontend, but you could consider coalescing the values here, too, and perhaps reduce the number of frontend checks.